proposals/Exceptions.md

```diff
@@ -170,21 +174,104 @@ also be sure to also pop off the caught exception values.
 
 ### Rethrows
 
-The _rethrow_ instruction has no arguments, and can only appear in catch
-blocks. It always re-throws the exception caught by the catch block. This allows
-the catch block to clean up state before the exception is passed back to the
-next enclosing try block.
+The _rethrow_ instruction can only appear in the body of a catch block.  The
+_rethrow_ instruction always re-throws the exception caught by the referenced
```

Comment:
Maybe change this to "caught by an enclosing catch block" here. Which exactly is detailed below.